api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -69,22 +70,57 @@ public interface Snapshot extends Serializable {
    *
    * @return a list of ManifestFile
    */
+  @Deprecated
   List<ManifestFile> allManifests();
 
+  // TODO - should we include this in the interface? Maybe keep the old ones in case the user knows the
+  //   values have been materialized / cached.
+  /**
+   * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
+   * <p>
+   * The FileIO instance is passed in for cases where the correct FileIO to use is not known when the
+   * TableMetadata (and thus BaseSnapshot implementations) is not known, such as parsing JSON for the
+   * RESTCatalog or for highly sessionzed applications like Trino or shared Spark servers.
+   *
+   * @param fileIO the FIleIO instance to be used for reading Snapshots, if not already
+   *        cached
+   * @return a list of ManifestFile
+   */
+  List<ManifestFile> allManifests(FileIO fileIO);
```

Comment:
Within `BaseSnapshot`, naming it `io` causes a checkstyle error because of the class level field `io`.

(ignore the line numbers)
```
BaseSnapshot.java:239:52: 'io' hides a field. [HiddenField]
```

We can keep the name as `io` within the interface though.

Reply:
I see. Makes sense for here, then. But let's update the interface like you said.